File: coderbyte/a_easy/11_ab_check/ab_check.rb
def ab_check(str)
  bools = []

  for n in 0..str.length-1 do
    if str[n] == 'a'
      if str[n + 4] == 'b'
        bools << true
      else
        bools << false
      end
    elsif str[n] == 'b'
      if str[n + 4] == 'a'
        bools << true
      else
        bools << false
      end
    end
  end

  bools.include? true
end

File: coderbyte/a_easy/javascript/11_ab_check/ab_check.js
const abCheck = (str) => {
  const array = str.split('');

  for (let i = 0; i < array.length; i++) {
    if (array[i] === 'a' && array[i + 4] === 'b' ||
      array[i] === 'b' && array[i + 4] === 'a') {
      return true;
    }
  }

  return false;
};

module.exports = abCheck;
